protocol/0092-SAMM-spot_automated_market_maker.md

```diff
@@ -0,0 +1,223 @@
+
+An automated market maker is configured at a per-key level, and is enabled by submitting a transaction with the requisite parameters. At this point in time the protocol will move committed funds to a sub-account which will be used to manage margin for the AMM. Once enabled, the configuration will be added to the pool of available AMMs to be utilised by the matching engine.
+
+Each party may have only one AMM configuration per market, and both Spot and Futures markets are eligible, with the behaviour differing slightly for each.
+
+## Process Overview
+
+The configuration and resultant lifecycle of an automated market maker is as follows:
+
+- Party funds a key which will be used by the strategy with desired token amounts.
+- Party submits a transaction containing configuration for the strategy on a given market. This will contain:
+  - Amount of base token to commit OR Amount of quote token to commit
+  - Price bounds (upper, lower)
+- Once accepted, the network will transfer funds to a sub-account and use the other parameters for maintaining the position.
+- The party can cancel the AMM at any time, with the spot balances immediately returned to their general accounts.
+
+## Sub-Account Configuration
+
+Each main Vega key will have one associated sub account for a given market, on which an AMM may be set up. The account key should be generated through a hash of the main account key plus the ID of the market to generate a valid Vega address in a predictable manner. Outside of the AMM framework the sub-accounts are treated identically to any other account, they will have the standard associated margin/general accounts and be able to place orders if required as with any other account. The key differentiator is that no external party will have the private key to control these accounts directly. The maintenance of such an account will be performed through a few actions:
+
+- Creation: A sub-account will be funded when a user configures an AMM strategy with a set of criteria and a commitment amount. At this point in time the commitment amount will be transferred to the sub-account's general account and the AMM strategy will commence
+- Cancellation: When the AMM is cancelled, balances are immediately returned to the user.
+- Amendment: Updates the strategy or commitment for a sub-account
+
+## Interface
+
+All AMM configurations should implement two key interfaces:
+
+- One taking simply the current state (`position` and `total funds`) and a trade (`volume`, `side`) and returning a quote price. This should also handle a trade of `volume = 0` to return a notional `fair price`
```

Comment:
Might just be me, but it's not obvious to me what `fair price` means. _I think_ its the price implied by the relative amounts of the two tokens but am not sure actually. Definiting it would make it a bit easier to follow.

Reply:
Adding an attempted clarification